logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -90,13 +94,21 @@ public DeadLetterQueueReader(Path queuePath, boolean cleanConsumed, SegmentListe
         this.segments = new ConcurrentSkipListSet<>(
                 Comparator.comparingInt(DeadLetterQueueUtils::extractSegmentId)
         );
-        segments.addAll(getSegmentPaths(queuePath).collect(Collectors.toList()));
+        segments.addAll(listSegmentPaths(queuePath).collect(Collectors.toList()));
         this.cleanConsumed = cleanConsumed;
         if (cleanConsumed && segmentCallback == null) {
             throw new IllegalArgumentException("When cleanConsumed is enabled must be passed also a valid segment listener");
         }
         this.segmentCallback = segmentCallback;
         this.lastConsumedReader = null;
+        if (cleanConsumed) {
+            // force single DLQ reader when clean consumed is requested
+            try {
+                fileLock = FileLockFactory.obtainLock(queuePath, "dlq_reader.lock");
+            } catch (LockException ex) {
+                throw new LockException("DLQ reader failed to obtain exclusive access, cause: " + ex.getMessage());
```

Comment:
I think maybe something more user friendly and actionable (either here or in the plugin).

```suggestion
                throw new LockException("Existing `dlg_reader.lock` file in <location>. Only one DeadLetterQueueReader with `cleanConsumed` set is allowed per Dead Letter Queue." + ex.getMessage());
```

Reply:
Fixed by d1f35e6c3dab19b48002f4b2dc01adb1c142c89f